src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java

```diff
@@ -81,7 +82,8 @@
       GuildFunctionData.of(
           (map, data) -> {
             Guild guild = map.get(data.getGuildId());
-            List<GuildMember> members = new ArrayList<>(guild.getMembers());
+            List<GuildMember> members =
+                new ArrayList<>(guild.getMembers().orElse(ImmutableList.of()));
```

Comment:
What you've said about orElse empty list makes sense in some places, but here isn't it going to delete members that are not in the list?

My understanding is that if it's not present, then we shouldn't be updating the member list.

Reply:
If it's not present don't you think we should make it present and give the members of the guild a piece in the cache? If the members are assigned to the guild and the members aren't currently present why would we dispose of them? If it's not present and it's updating members at all we should know about the members. Also, I'm using the guild which is cached, they don't send a guild through along with the update. If the guild which is cached already doesn't have the members they don't exist to us.

Reply:
My issue is the kind of scenario where:
* We have guild members cached
* We receive an event where the list of members is null
* We update the cache of guild members we have to empty list

Reply:
I don't see a case where this is relevant. I never use the given list of members as the list we use in the cache. I always use the list as a reference of what to update. If the list is empty it updates nothing.

Reply:
Is this code here removing members based on what's in the cache and what's in the request? https://github.com/CoreyShupe/ErisCasper.java/blob/7118706dd4539a14f96befac1e30451512e7aa41/src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java#L101-L113

Reply:
What this does is it takes the given list of items, takes the list turns it into a stream, and translates the active members into removable members if they appear in the container of removed snowflakes, then once it finds all of the snowflakes contained in the cache it removes the member version of them. It removes cached members based off information given in the request, not vice versa. If the list is empty nothing will happen because the container is empty. It's like querying an empty set if it has an item or not. It just returns no then moves on, therefore no members would actually be processed and the Stack would remain empty.

Reply:
I think I worded that properly.

Reply:
ok, cool

Reply:
Also, once it removes the current set of members shown in the stack it adds all of the members back (as their updated version, just to stop members from showing up twice). I think there's actually a different error which occurs just due to object injection, which I'll fix.

Reply:
So I actually worded that a bit wrong. End of the day, the members inside of the data get pushed into the cache, if empty nothing is changed, if there's stuff to remove it always just updates the cache. It never replaces the cache.